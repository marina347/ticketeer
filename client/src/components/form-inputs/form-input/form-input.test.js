import React from "react";
import { shallow } from "enzyme";

import FormInput from "./form-input.component";

describe("FormInput component", () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();
    const mockProps = {
      label: "name",
      containerClassName: "class-1",
      handleChange: mockHandleChange,
    };
    wrapper = shallow(<FormInput {...mockProps} />);
  });

  it("Should render FormInput component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should call handleChange method when input changes", () => {
    wrapper.find('[id="form-input"]').simulate("change");
    expect(mockHandleChange).toHaveBeenCalled();
  });

  it("Should render form label", () => {
    expect(wrapper.exists('[id="form-input-label"]')).toBe(true);
  });

  it("Should not render form label", () => {
    const mockNewProps = {
      label: "",
      containerClassName: "class-1",
      handleChange: mockHandleChange,
    };

    const newWrapper = shallow(<FormInput {...mockNewProps} />);
    expect(newWrapper.exists('[id="form-input-label"]')).toBe(false);
  });
});
