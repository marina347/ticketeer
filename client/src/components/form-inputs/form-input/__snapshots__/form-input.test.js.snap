// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component Should render FormInput component 1`] = `
<div
  className="form-input"
>
  <input
    className="form-input__input"
    id="form-input"
    onChange={[MockFunction]}
    required={true}
  />
  <label
    className="form-input__label"
    id="form-input-label"
  >
    name
  </label>
</div>
`;
